feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -83,13 +90,21 @@ class BillingWrapper(
         }
     }
 
+    override fun startConnectionOnMainThread(delayMilliseconds: Long) {
+        mainHandler.postDelayed(
+            { startConnection() },
+            delayMilliseconds
+        )
+    }
+
     override fun startConnection() {
-        mainHandler.post {
-            synchronized(this@BillingWrapper) {
-                if (billingClient == null) {
-                    billingClient = clientFactory.buildClient(this)
-                }
-                billingClient?.let {
+        synchronized(this@BillingWrapper) {
+            if (billingClient == null) {
+                billingClient = clientFactory.buildClient(this)
+            }
+
+            billingClient?.let {
+                if (!it.isReady) {
```

Comment:
I thought we wanted to check here if it's connecting? to not call `startConnection` a lot of times

Reply:
`ConnectionState` was added in BillingClient4, so checking it was causing the unityIap flavor build to fail. my original concern with calling startConnection multiple times is that we might get multiple `onBillingSetupFinished(OK)` callbacks. so i tested that out, and it turns out that BillingClient instead immediately returns onBillingSetupFinished(DEVELOPER_ERROR) to indicate that it's already trying to connect, and then onBillingSetupFinsihed(OK) when done. we have this in our switch statement for the Billing result code:

``` 
BillingClient.BillingResponseCode.DEVELOPER_ERROR -> {
        // Billing service is already trying to connect. Don't do anything.
}
```

if we still want to avoid extra calls, i think we'd need to keep track of a connection state ourselves.